plugin/pkg/admission/resourcequota/controller.go

```diff
@@ -436,7 +491,11 @@ func (e *quotaEvaluator) checkRequest(quotas []api.ResourceQuota, a admission.At
 	}
 
 	if len(interestingQuotaIndexes) == 0 {
-		return quotas, nil
+		if objectRequiresQuotaStrictly {
+			return quotas, fmt.Errorf("No matching quota found with the configured MatchScopes")
```

Comment:
i am not sure i understand the logic here.

i would expect that this would be handled similar to the hasNoCoveringQuota section above.

the error message could be improved to say "insufficient quota to consume: <resources> that match following scopes: <scopes>"